### PR TITLE
metric input

### PR DESCRIPTION
**Pull request by ewojtach**

### CODE REVIEW

**Review by mikouaj — COMMENTED** [review]
Good work, please continue :)

File: internal/config/config.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ type Config struct {
 	CredentialsFile  string                 `yaml:"credentialsFile"`
 	Clusters         []ConfigCluster        `yaml:"clusters"`
 	Policies         []ConfigPolicy         `yaml:"policies"`
-	Inputs           []ConfigInput          `yaml:"inputs"`
+	Inputs           ConfigInput            `yaml:"inputs"`
```
[the remark this answers is not in the document]
Comment (ewojtach) [reply internal/config/config.go:47]: ConfigInput structure consists of multiple fields related to particular input structure

[the remark this answers is not in the document]
Comment (mikouaj) [reply internal/config/config.go:47]: Got it, thanks!

File: internal/inputs/clients/metrics.go [hunk 2]
```diff
@@ -69,10 +69,52 @@ func NewMetricClient(ctx context.Context, projectId string, authToken string) (M
...
+}
+
+func (b *metricsClientBuilder) Build() (MetricsClient, error) {
+
+	var maxGoRoutines = defaultMaxGoroutines
+	if b.maxGoroutines != 0 {
+		maxGoRoutines = b.maxGoroutines
+	}
+
+	metricsClient, err := newMetricsClient(b.ctx, b.projectId, b.authToken, maxGoRoutines)
```
Comment (mikouaj) [inline internal/inputs/clients/metrics.go:109]: We will need to find way to add our customer User Agent header to the Cloud Monitoring prometheus queries (for usage tracking purposes).
I couldn't find anything in prometheus API unfortunately.. The last resort solution seems to be crating custom `RoundTripper` adapter over the `AuthorizationCredentialsRoundTripper` and add header there on roundtrip.

**Review by mikouaj — COMMENTED** [review]
Looks good !:) I just can' see any, even simple tests for metric_input :/

**Review by mikouaj — APPROVED** [review]
LGTM! :-)

**Pull request merged**